File: find_pairs/findPairs.js
function findPairs(arr, target) {
  const indicesMap = new Map();
  arr.forEach((num, index) => {
    const complement = target - num;
    if (indicesMap.has(complement)) {
      console.log(
        `Pair found at ${indicesMap.get(
          complement
        )} and ${index} (${complement} + ${num})`
      );
    }
    indicesMap.set(num, index);
  });
}

// Sample usage
const arr = [1, 3, 5, 7, 2, 4];
const target = 8;
findPairs(arr, target);

File: find_pairs/find_pairs.py
def find_pairs(arr, target):
    indices_map = {}
    for index, num in enumerate(arr):
        complement = target - num
        if complement in indices_map:
            print(f"Pair found at {indices_map[complement]} and {index} ({complement} + {num})")
        indices_map[num] = index

# Sample input
arr = [1, 3, 5, 7, 2, 4]
target = 8
find_pairs(arr, target)
